--- unixbench-4.1.0/pgms/tst.sh.orig	Mon Nov 27 18:07:03 1995
+++ unixbench-4.1.0/pgms/tst.sh	Wed Oct 24 11:53:23 2001
@@ -14,7 +14,7 @@
 #
 ###############################################################################
 ID="@(#)tst.sh:3.4 -- 5/15/91 19:30:24";
-sort >sort.$$ <sort.src
+sort >sort.$$ <${UB_BINDIR}/sort.src
 od sort.$$ | sort -n -k 1 > od.$$
 grep the sort.$$ | tee grep.$$ | wc > wc.$$
 rm sort.$$ grep.$$ od.$$ wc.$$
--- UnixBench/Run.orig	2012-06-05 14:41:53.016735725 +0200
+++ UnixBench/Run	2012-06-05 14:42:19.356633000 +0200
@@ -434,7 +434,7 @@
         "logmsg" => "C Compiler Throughput ($cCompiler)",
         "cat"    => 'misc',
         "prog" => "${BINDIR}/looper",
-        "options" => "60 $cCompiler cctest.c",
+        "options" => "60 $cCompiler ${BINDIR}/cctest.c",
     },
     "arithoh" => {
         "logmsg" => "Arithoh",
@@ -471,7 +471,7 @@
         "cat"    => 'misc',
         "prog" => "${BINDIR}/looper",
         "options" => "30 dc",
-        "stdin" => "dc.dat",
+        "stdin" => "${BINDIR}/dc.dat",
     },
     "hanoi" => {
         "logmsg" => "Recursion Test -- Tower of Hanoi",
@@ -482,7 +482,7 @@
         "logmsg" => "Grep a large file (system's grep)",
         "cat"    => 'misc',
         "prog" => "${BINDIR}/looper",
-        "options" => "30 grep -c gimp large.txt",
+        "options" => "30 grep -c gimp ${BINDIR}/large.txt",
     },
     "sysexec" => {
         "logmsg" => "Exec System Call Overhead",
@@ -784,15 +784,6 @@
     # Set the language.
     $ENV{'LANG'} = $language;
 
-    # Check that the required files are in the proper places.
-    system("make check");
-    if ($? != 0) {
-        system("make all");
-        if ($? != 0) {
-            abortRun("\"make all\" failed");
-        }
-    }
-
     # Create a script to kill this run.
     system("echo \"kill -9 $$\" > \"${TMPDIR}/kill_run\"");
     chmod(0755, $TMPDIR . "/kill_run");
