--- unixbench-4.1.0/pgms/tst.sh.orig	Mon Nov 27 18:07:03 1995
+++ unixbench-4.1.0/pgms/tst.sh	Wed Oct 24 11:53:23 2001
@@ -14,7 +14,7 @@
 #
 ###############################################################################
 ID="@(#)tst.sh:3.4 -- 5/15/91 19:30:24";
-sort >sort.$$ <sort.src
+sort >sort.$$ <${UB_BINDIR}/sort.src
 od sort.$$ | sort -n -k 1 > od.$$
 grep the sort.$$ | tee grep.$$ | wc > wc.$$
 rm sort.$$ grep.$$ od.$$ wc.$$
--- UnixBench/Run.orig	2012-06-05 14:41:53.016735725 +0200
+++ UnixBench/Run	2012-06-05 14:42:19.356633000 +0200
@@ -784,15 +784,6 @@
     # Set the language.
     $ENV{'LANG'} = $language;
 
-    # Check that the required files are in the proper places.
-    system("make check");
-    if ($? != 0) {
-        system("make all");
-        if ($? != 0) {
-            abortRun("\"make all\" failed");
-        }
-    }
-
     # Create a script to kill this run.
     system("echo \"kill -9 $$\" > \"${TMPDIR}/kill_run\"");
     chmod(0755, $TMPDIR . "/kill_run");
