diff -Nur unixbench-4.1.0-orig/Makefile unixbench-4.1.0/Makefile
--- unixbench-4.1.0-orig/Makefile	1999-07-28 22:05:02.000000000 +0000
+++ unixbench-4.1.0/Makefile	2006-02-25 18:08:52.324873250 +0000
@@ -64,7 +64,7 @@
 TMPDIR = ./tmp
 # other directories
 INCLDIR = /usr/include
-LIBDIR = /lib
+LIBDIR = /lib64
 SCRIPTS = unixbench.logo cleanup.sh \
 	fs.awk multi.sh tst.sh index.sh index.base \
 	index.awk loops.awk loopm.awk mwips.awk \
diff -Nur unixbench-4.1.0-orig/Run unixbench-4.1.0/Run
--- unixbench-4.1.0-orig/Run	2006-02-25 17:50:23.776399000 +0000
+++ unixbench-4.1.0/Run	2006-02-25 18:09:38.467757000 +0000
@@ -92,7 +92,7 @@
 
 trap "rm -rf $WORKDIR; exit" EXIT INT
 
-UNIXBENCHDIR="/usr/lib/unixbench"
+UNIXBENCHDIR="/usr/lib64/unixbench"
 
 BINDIR=${UNIXBENCHDIR}
 # let's extend the path to this directory
